egld-esdt-swap/src/lib.rs

```diff
@@ -82,24 +105,75 @@ pub trait EgldEsdtSwap {
 
     // private
 
-    fn data_or_empty(&self, to: &Address, data: &'static [u8]) -> &[u8] {
-        if self.blockchain().is_smart_contract(to) {
-            &[]
-        } else {
-            data
-        }
+    fn needs_execution(&self, caller: &Address, function: &BoxedBytes) -> bool {
+        self.blockchain().is_smart_contract(caller) && !function.is_empty()
+    }
+
+    fn transfer_and_execute_via_async(
```

Comment:
Could also rename to something like transfer_via_async, since transfer & execute is a completely different function (even if they share some functionality)

Reply:
Done